grpc-ingest/src/download_metadata.rs

```diff
@@ -0,0 +1,236 @@
+use {
+    crate::{
+        config::{ConfigDownloadMetadata, ConfigDownloadMetadataOpts},
+        postgres::{create_pool as pg_create_pool, metrics_pgpool},
+        util::create_shutdown,
+    },
+    digital_asset_types::dao::{asset_data, sea_orm_active_enums::TaskStatus, tasks},
+    futures::{
+        future::{pending, FutureExt},
+        stream::StreamExt,
+    },
+    program_transformers::DownloadMetadataInfo,
+    reqwest::{ClientBuilder, StatusCode},
+    sea_orm::{
+        entity::{ActiveValue, ColumnTrait, EntityTrait},
+        query::{Condition, Order, QueryFilter, QueryOrder, QuerySelect},
+        sea_query::expr::Expr,
+        SqlxPostgresConnector, TransactionTrait,
+    },
+    sqlx::PgPool,
+    std::{sync::Arc, time::Duration},
+    tokio::{
+        sync::{mpsc, Notify},
+        task::JoinSet,
+        time::sleep,
+    },
+    tracing::{info, warn},
+};
+
+pub const TASK_TYPE: &str = "DownloadMetadata";
+
+pub async fn run(config: ConfigDownloadMetadata) -> anyhow::Result<()> {
```

Comment:
Related to my previous question about how many instances of each process running.  Is it safe to assume that right now there's only one instance of each type of task?  Otherwise seems like there could be race conditions such as one process is resetting tasks to `Pending` or `Running` and another process is trying to execute tasks, etc.

Another interesting option with a rearchitecture like this, you could also consider putting the download metadata tasks into Redis.

Reply:
> Another interesting option with a rearchitecture like this, you could also consider putting the download metadata tasks into Redis.

Agreed. As part of Triton running this in prod we are going to do just that based on metadata-json crate we are running now but will be combining with grpc-ingest.

https://github.com/rpcpool/digital-asset-rpc-infrastructure/tree/triton-build/metadata_json

I am also going to look at adding some metadata to asset data to help skip dead links. Idea is to add last_requested_status, requested_attempts.